### DISCUSSION TITLE
derive(DeriveEntityModel) generate new type in proc macro is not friendly to intellij-rust static analysis

### DISCUSSION THREAD

**Discussion by pymongo** [SeaQL/sea-orm#601] [General]
Intellij-rust can't recognize the type define in proc macro, sea-orm-cli generte code is not friendly to intellij-rust, generate/define new type in macro is not friendly to IDE

![image](https://user-images.githubusercontent.com/42347370/158048380-316e7720-b724-41a2-bc09-7626f8a5c7b0.png)

## Addition

rust-analyzer can resolve the type generate in proc macro

**Comment by tyt2y3** [comment]
https://github.com/intellij-rust/intellij-rust/issues/8610

**Reply by pymongo** [reply]
Thanks!
